### DISCUSSION TITLE
Share volumes between GitHub Actions jobs

### DISCUSSION THREAD

**Discussion by slavafomin** [community/community#5379] [Actions]
When working with GitHub Actions it's very convenient to have multiple jobs to perform various tasks. This allows to create more complex and flexible workflows where some jobs are performed concurrently while others are being awaited for. This also helps to improve the performance of the workflow by parallelizing the work.

However, right now, there is no easy way to share files between the jobs, because each job is a separate VM. But in order to perform some meaningful work it's almost always necessary to work on the same set of files or to pass files from one job to another.

The lack of such functionality forces developers to use a single job for the entire workflow, which reduces the convenience, flexibility and possible the performance.

The only workaround right now is to use the **artifacts** or **cache**, but these features are implemented in a way that has a very poor performance (because files are need to be compressed and sent over HTTP to an external server and then to be downloaded again).

If GitHub Actions would provide a way to share volumes (filesystems) between the jobs it would allow to overcome all these limitations in a much more performant and convenient way.

**Comment by jsoref** [comment]
Technically you can also uses `outputs` to share things. I'm not sure offhand what the restriction on size is.

But note that often jobs will be queued on distinct instances, so you do need some sort of remoting system...
